uaclient/clouds/identity.py

```diff
@@ -49,7 +49,33 @@ def get_cloud_type() -> Tuple[Optional[str], Optional[NoCloudTypeReason]]:
     return (None, NoCloudTypeReason.NO_CLOUD_DETECTED)
 
 
+class CloudFactoryError(Exception):
+    def __init__(self, cloud_type: Optional[str]) -> None:
+        self.cloud_type = cloud_type
+
+
+class CloudFactoryNoCloudError(CloudFactoryError):
+    pass
+
+
+class CloudFactoryUnsupportedCloudError(CloudFactoryError):
+    pass
+
+
+class CloudFactoryNonViableCloudError(CloudFactoryError):
+    pass
```

Comment:
Maybe we should move those into the `exceptions` module ?

Reply:
Good point, will do